### PR TITLE
refactor: use cli framework and handlers structure

### PR DESCRIPTION
**Pull request by AH-dark**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,29 +3,37 @@ package main
...
+	"github.com/afadian/fadian-go/entry"
 )
 
 var (
-	app *fx.App
 	ctx = context.Background()
+	app *fx.App
+
+	version = "0.0.1"
+	commit  = "0000000"
```
Comment (purofle) [inline main.go:18]: commit 要不要改成这个 commit 的 hash

File: cli/app.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+func InvokeApp(lc fx.Lifecycle, params Params) {
+	app := &cli.App{
+		Name:      "fadian-go",
+		Usage:     "发癫",
+		Version:   fmt.Sprintf("%s-%s", params.Version, params.Commit),
+		Copyright: fmt.Sprintf("2022-%d © 爱发癫 All Rights Reserved", time.Now().Year()),
+		Authors: []*cli.Author{
+			{Name: "AH Dark", Email: "ahdark@outlook.com"},
+			{Name: "Kevin Williams", Email: ""}, // TODO: add email
+			{Name: "Purofle", Email: ""},        // TODO: add email
```
Comment (purofle) [inline cli/app.go:32]:
```
{Name: "AH Dark", Email: "ahdark@outlook.com"},
{Name: "Kevin Williams", Email: "admin@utermux.dev"},
{Name: "Purofle", Email: "purofle@gmail.com"},
```

**Review by purofle — CHANGES_REQUESTED** [review]

**Review by 2096779623 — CHANGES_REQUESTED** [review]
```
~/fadian-go $ ./fadian-go
NAME:
   fadian-go - 发癫

USAGE:
   fadian-go [global options] command [command options] [arguments...]

VERSION:
   0.0.1-0000000

AUTHORS:
   AH Dark <ahdark@outlook.com>
   Kevin Williams
   Purofle

COMMANDS:
   fadian
   fabing
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --debug, -d    是否开启调试模式 (default: false) [$DEBUG]
   --num value    重复次数 (default: 1)
   --help, -h     show help
   --version, -v  print the version

COPYRIGHT:
   2022-2023 © 爱发癫 All Rights Reserved
~/fadian-go $
```
更新一下README.md

**Comment by 2096779623** [comment]
```
~/fadian-go $ ./fadian-go -v
flag provided but not defined: -v
Usage of ./fadian-go:
  -d    是否开启调试模式
~/fadian-go $ ./fadian-go -h
Usage of ./fadian-go:
  -d    是否开启调试模式
~/fadian-go $ ./fadian-go --version
flag provided but not defined: -version
Usage of ./fadian-go:
  -d    是否开启调试模式
~/fadian-go $
```
~~好像~~有bug

**Comment by 2096779623** [comment]
```
~/fadian-go $ ./fadian-go fabing
NAME:
   fadian-go fabing

USAGE:
   fadian-go fabing [command options] [arguments...]

OPTIONS:
   --name value, -n value  name to be fabing
   --help, -h              show help
panic: Required flag "name" not set; sync /dev/stderr: invalid argument

goroutine 1 [running]:
main.main()
        /data/data/com.termux/files/home/fadian-go/main.go:35 +0x64
~/fadian-go $ ./fadian-go fabing -n a
~/fadian-go $ ./fadian-go fadian -n a
我好想做 a 小姐的狗啊。
```
fabing seems dosen't work.

File: main.go (hunk 1 above)
Comment (AH-dark) [reply main.go:18]: 没弄完呢，这个会用 `-ldflags` 在每次 build 的时候指定

**Comment by AH-dark** [comment]
> ```
> ~/fadian-go $ ./fadian-go -v
> flag provided but not defined: -v
> Usage of ./fadian-go:
>   -d    是否开启调试模式
> ~/fadian-go $ ./fadian-go -h
> Usage of ./fadian-go:
>   -d    是否开启调试模式
> ~/fadian-go $ ./fadian-go --version
> flag provided but not defined: -version
> Usage of ./fadian-go:
>   -d    是否开启调试模式
> ~/fadian-go $
> ```
> 
> ~好像~有bug

Fixed. This seems to be a conflict between flags package and cli framework.

**Comment by AH-dark** [comment]
@2096779623 Now you need to use such a command to release a new tag, CI will help you generate a draft release.

```bash
git tag v1.0.0
git push --tags
```

**Review by 2096779623 — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (purofle) [reply main.go:16]: 还有这个没改

Comment (AH-dark) [reply main.go:16]: 改了，你看我发群里的 snapshot

**Review by purofle — APPROVED** [review]

**Pull request merged**